### DISCUSSION TITLE
how to repeat the last trial?

### DISCUSSION THREAD

**Discussion by neuronflow** [jspsych/jsPsych#674] [Q&A] [answered]
I am collecting responses from users. If they press the wrong button I want them to be able to correct their response and press the backspace button. How can I implement this?

I think I somehow need to make a copy of the current trial and the trial before that I want to repeat and inject it in the timeline?

After answering with the corrected response the experiment should then resume with the next trial and repeat the copies of "current trial" and the "to be corrected trial" at the end of the stimulus block.

**Comment by jodeleeuw** [comment, marked as answer]
> I think I somehow need to make a copy of the current trial and the trial before that I want to repeat and inject it in the timeline?

One way to do this is to use [a looping timeline](https://www.jspsych.org/overview/timeline/#looping-timelines). You can put the trials on loop, and only proceed when the user answers correctly.

> After answering with the corrected response the experiment should then resume with the next trial and repeat the copies of "current trial" and the "to be corrected trial" at the end of the stimulus block.

This is a little trickier. There is a function to [add another trial to the end of the timeline](https://www.jspsych.org/core_library/jspsych-core/#jspsychaddnodetoendoftimeline), but that may not be ideal if your sequence is in the middle of the experiment.

Another option is to use a looping timeline with dynamic parameters and control the set of parameters outside of jsPsych. I recently recorded a [video tutorial](https://www.youtube.com/watch?v=fYKaKTvybCI) that does something kind of similar, but in a very different context. It might help you see how to do this sort of thing though.
